src/util.rs

```diff
@@ -238,3 +239,25 @@ impl U32Ext for u32 {
         })
     }
 }
+
+/// Return only the clusters from the slice of either register or clusters.
+pub fn only_clusters(ercs: &[Either<Register, Cluster>]) -> Vec<&Cluster> {
```

Comment:
off-topic: this looks like something that could be provided by the either crate. It looks a bit like [`Iterator::partition`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.partition) but that method returns two `Vec`s of the same type.

Reply:
I opened an issue in bluss/either#20 regarding this

Reply:
At least right now, I won't be implementing this, because I can't figure out how to get itertools to figure out the vec of refs. Open to PRs in the future :)